x/posts/client/cli/tx.go

```diff
@@ -84,6 +104,80 @@ func GetCmdCreatePost(cdc *codec.Codec) *cobra.Command {
 				return err
 			}
 
+			// medias' checks
+
+			mediasStrings, _ := cmd.Flags().GetStringArray(flagMedia)
+			medias := types.PostMedias{}
+			for _, mediaString := range mediasStrings {
+				argz := strings.Split(mediaString, ",")
+				if len(argz) != 2 {
+					return fmt.Errorf("if medias are specified, the arguments has to be exactly 2 and in this order: \"URI,Mime-Type\", please use the --help flag to know more")
+				}
+
+				media := types.NewPostMedia(argz[0], argz[1])
+				medias = medias.AppendIfMissing(media)
+			}
+
+			// polls' checks
+
+			pollDetailsMap := viper.GetStringMapString(flagPollDetails)
+			pollAnswersSlice := viper.GetStringSlice(flagPollAnswer)
+
+			if len(pollDetailsMap) == 0 && len(pollAnswersSlice) > 0 {
+				return fmt.Errorf("poll answers specified but no poll details found. Please use %s to specify the poll details", flagPollDetails)
+			}
+
+			if len(pollDetailsMap) > 0 && len(pollAnswersSlice) == 0 {
+				return fmt.Errorf("poll details specified but answers are not. Please use the %s to specify one or more answer", flagPollAnswer)
+			}
+
+			var pollData *types.PollData = nil
```

Comment:
No need to initialize to `nil` as far as I know

```suggestion
			var pollData *types.PollData
```